File: 1198-find-smallest-common-element-in-all-rows/1198-find-smallest-common-element-in-all-rows.cpp
class Solution {
public:
    int smallestCommonElement(vector<vector<int>>& mat) {
        int m = mat.size(), n = mat[0].size();
        int count[10001] = {0};
        for (int i = 0; i < m; ++i) {
            for (int j = 0; j < n; ++j) {
                count[mat[i][j]]++;
                if (count[mat[i][j]] == m) return mat[i][j];
            }
        }
        return -1;
    }
};

File: 1198-find-smallest-common-element-in-all-rows/1198-find-smallest-common-element-in-all-rows.js
/**
 * @param {number[][]} mat
 * @return {number}
 */
var smallestCommonElement = function(mat) {
    var m = mat.length;
    var n = mat[0].length;
    var c = new Array(10001).fill(0);
    for (let i=0; i<m; ++i) {
        for (let j=0; j<n; ++j) {
            c[mat[i][j]]++;
            if (c[mat[i][j]] == m) return mat[i][j];
        }
    }
    return -1;
};